regression/goto-analyzer/ternary-operator-constants/main.c

```diff
@@ -0,0 +1,19 @@
+#include <stdbool.h>
+
+extern int x;
+
+int main(void)
+{
+  bool b1;
+  bool b2;
+
+  b1 = true;
+  b2 = !b1;
+
+  bool b3 = x ? true : false;
+  int i = b1 ? 10 : 20;
+  int j = b2 ? 10 : 20;
+  int k = b3 ? 10 : 20;
+
```

Comment:
If the test is demonstrating what values are represented by the abstract domain then I have a mild preference for having `assert`s in the test cases, ideally passing and failing.  I tend to use `--show` in tests for when I want to actually check the displayed output.  This is not wrong and you don't need to change it, more of a point of style really.

Reply:
I wasn't sure how to write an assert that would show that the value-set did indeed contain all the values I expected it to.

Reply:
It's a bit of a faff so I wouldn't require it but the following should give you the exact values in the value set via the assertion checking results:

```
if (k == value1) { assert(1); }
else if (k == value2) { assert(1); }
...
else (k == valueN) { assert(1); }
else { assert(0); }
```

If you have enumerated all of the values then each of the `assert(1)` should be marked as a success.  If it is unreachable then the value has been omitted.  Correspondingly, if `assert(0)` is unreachable then there are no extra values, if it fails then there are additional, possibly unexpected, values.

Ofcourse, having written that, I am not 100% sure that `abstract_environmentt::assume()` will work well enough with `value_sett` for this to work.  At some point it should but that is not necessary for this PR.

Reply:
Thanks. Noted for the future.